server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -268,5 +277,28 @@ protected double computeCost(
     return bestServer;
   }
 
+  @Override
+  public boolean steady(final List<ServerHolder> serverHolders)
+  {
+    if (balancerSlope <= BalancerStrategy.DEFAULT_BALANCER_SLOPE) {
+      return false;
+    }
+    double totalCost = 0;
+    double maxCost = Double.MIN_VALUE;
+    double minCost = Double.MAX_VALUE;
+    for (ServerHolder server : serverHolders) {
+      double serverCost = calculateServerCost(server);
+      totalCost += serverCost;
+      maxCost = Math.max(maxCost, serverCost);
+      minCost = Math.min(minCost, serverCost);
+    }
+    double average = totalCost / serverHolders.size();
+    double floor =  average * (1 - balancerSlope);
+    double ceiling =  average * (1 + balancerSlope);
+    if (maxCost < ceiling && minCost > floor) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
does this assume that all your historicals are of same capacity?
given that cost typically goes with number of segments on the historicals... this will produce higher cost for higher capacity historicals and will pretty much always be false.